Client/Frontend/Browser/HomePanel/NTPDownloader.swift

```diff
@@ -0,0 +1,378 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Shared
+
+private let logger = Logger.browserLogger
+
+struct NTPFocalPoint: Codable {
+    let x: Int
+    let y: Int
+}
+
+struct NTPLogo: Codable {
+    let imageUrl: String
+    let alt: String
+    let companyName: String
+    let destinationUrl: URL
+}
+
+struct NTPWallpaper: Codable {
+    let imageUrl: String
+    let focalPoint: NTPFocalPoint?
+}
+
+struct NTPItemInfo: Codable {
+    let logo: NTPLogo
+    let wallpapers: [NTPWallpaper]
+}
+
+class NTPDownloader {
+    private static let etagFile = "crc.etag"
+    private static let metadataFile = "photo.json"
+    private static let ntpDownloadsFolder = "NTPDownloads"
+    private static let baseURL = "https://brave-ntp-crx-input-dev.s3-us-west-2.amazonaws.com/"
+    
+    private let isZipped: Bool
+    private let defaultLocale = "US"
+    private let supportedLocales: [String] = []
+    private let currentLocale = Locale.current.regionCode
+    
+    init(isZipped: Bool) {
+        self.isZipped = isZipped
+    }
+    
+    func getNTPInfo(_ completion: @escaping (NTPItemInfo?) -> Void) {
+        // Download the NTP Info to a temporary directory
+        self.download { [weak self] url, cacheInfo, error in
+            guard let self = self else { return }
+            
+            if let error = error {
+                logger.error(error)
+                return completion(self.loadNTPInfo())
```

Comment:
Why not just return nil for these error paths?
Returning the same closure for each cases looks weird

Reply:
It must load from the cache the NTPInfo when an error has occurred.